components/ModFacilitySearchbar.vue

```diff
@@ -0,0 +1,181 @@
+import SVGCheckMark from '~/assets/icons/check-mark.svg'
+import SVGLookingGlass from '~/assets/icons/looking-glass.svg'
+import { useFacilitiesStore } from '#build/imports'
+import { useFacilitySearchbarStore } from '~/stores/facilitySearchbarStore'
+import type { Facility } from '~/typedefs/gqlTypes'
+
+//The limit for fetching facilities is 400 at the current date (30/11/2024)
+const facilitiesStore = useFacilitiesStore()
+const facilitySearchbarStore = useFacilitySearchbarStore()
+
+const searchInput = ref('')
+const searchInputElement = ref<HTMLInputElement | null>(null)
+const filteredFacilities = ref<Facility[]>([])
+const selectedFacilities = ref(facilitySearchbarStore.selectedFacilities)
+const selectedFacilityIndex = ref(0)
+
+const handleListItemClick = (facility: Facility) => {
+    searchInputElement.value?.focus()
+
+    if (selectedFacilities.value.has(facility)) {
+        selectedFacilities.value.delete(facility)
+        return
+    }
+
+    selectedFacilities.value.add(facility)
+}
+
+const handleSearchInputBlur = () => {
+    searchInput.value = ''
+}
+
+const handleSearchInputEsc = () => {
+    handleSearchInputBlur()
+}
+
+const handleSearchInputArrowUp = (event: KeyboardEvent) => {
+    // Prevent the cursor to be moving inside the search input
+    event.preventDefault()
+    if (selectedFacilityIndex.value > -1) selectedFacilityIndex.value -= 1
+}
+
+const handleSearchInputArrowDown = (event: KeyboardEvent) => {
+    event.preventDefault()
+    if (selectedFacilityIndex.value < filteredFacilities.value.length - 1) selectedFacilityIndex.value += 1
+}
+
+const handleSearchInputEnter = () => {
+    handleListItemClick(filteredFacilities.value[selectedFacilityIndex.value])
+}
+
+const handleSearchInputChange = () => {
+    const currentFacilities = facilitiesStore.facilityData
+    const facilityIdOrName = searchInput.value.toLowerCase()
+
+    selectedFacilityIndex.value = 0
+
+    if (facilityIdOrName.trim() === '') {
+        filteredFacilities.value = []
+        return
+    }
+
+    filteredFacilities.value = currentFacilities.filter(({ nameEn, nameJa, id }) => {
+        const isMatch
+        = nameEn.toLowerCase().includes(facilityIdOrName)
+        || nameJa.toLowerCase().includes(facilityIdOrName)
+        // TODO: Is the id case sensitive?
+        || id.toLowerCase() === facilityIdOrName
+        return isMatch
+    })
+}
+
+// TODO: Clear the store to prevent bugs. This might change in the future.
+
+onMounted(() => {
+    selectedFacilities.value = new Set()
+})
+
+onUnmounted(() => {
+    selectedFacilities.value = new Set()
+})
+
+watch(searchInput, handleSearchInputChange)
+
+// Displays the dropdown above the searchInputElement if its position is greater than window.innerHeight / 2
+const computedDropdownClass = computed(
+    () => {
+        if (searchInputElement.value && searchInputElement.value.getBoundingClientRect().bottom >= window.innerHeight / 2) {
+            return '-translate-y-full -top-14'
```

Comment:
Is this going to work for mobile? instead of `top-14` can we use something that's responsive?